src/standalone/Vegetation/pmodel.jl

```diff
@@ -0,0 +1,305 @@
+the model will work correctly. By default, these constants are set using the Pa convention. 
+$(DocStringExtensions.FIELDS)
+"""
+Base.@kwdef struct PModelConstants{FT}
+    """Gas constant (J mol^-1 K^-1)"""
+    R::FT
+    """Michaelis-Menten parameter for carboxylation at 25°C (mol mol^-1 or Pa)"""
+    Kc25::FT
+    """Michaelis-Menten parameter for oxygenation at 25°C (mol mol^-1 or Pa)"""
+    Ko25::FT
+    """Reference temperature equal to 25˚C (K)"""
+    To::FT
+    """Energy of activation for Kc (J mol^-1)"""
+    ΔHkc::FT
+    """Energy of activation for Ko (J mol^-1)"""
+    ΔHko::FT
+    """Relative diffusivity of CO2 in the stomatal pores, equal to 1.6."""
+    Drel::FT
+    """Effective energy of activation for Γstar (J mol^-1)"""
+    ΔHΓstar::FT
+    """Γstar at 25 °C (mol mol^-1 or Pa)"""
+    Γstar25::FT
+    """Effective energy of activation for Vcmax (J mol^-1)"""
+    Ha_Vcmax::FT
+    """Effective energy of deactivation for Vcmax (J mol^-1)"""
+    Hd_Vcmax::FT
+    """Intercept term for dS in Vcmax deactivation factor (J K^-1 mol^-1)"""
+    aS_Vcmax::FT
+    """Slope term for dS in Vcmax deactivation factor (J K^-2 mol^-1)"""
+    bS_Vcmax::FT
+    """Effective energy of activation for Jmax (J mol^-1)"""
+    Ha_Jmax::FT
+    """Effective energy of deactivation for Jmax (J mol^-1)"""
+    Hd_Jmax::FT
+    """Intercept term for dS in Jmax deactivation factor (J K^-1 mol^-1)"""
+    aS_Jmax::FT
+    """Slope term for dS in Jmax deactivation factor (J K^-2 mol^-1)"""
+    bS_Jmax::FT
+    """Molar mass of carbon (kg mol^-1)"""
+    Mc::FT
+    """Intercellular O2 mixing ratio (mol mol^-1)"""
+    oi::FT
+    """First order coefficient for temp-dependent Rd (K^-1)"""
+    aRd::FT
+    """Second order coefficient for temp-dependent Rd (K^-2)"""
+    bRd::FT
+    """Constant factor appearing the dark respiration term for C3 plants (unitless)"""
+    fC3::FT
+end
+
+Base.eltype(::PModelParameters{FT}) where {FT} = FT
+Base.eltype(::PModelConstants{FT}) where {FT} = FT
+
+Base.broadcastable(m::PModelParameters) = tuple(m)
+Base.broadcastable(m::PModelConstants) = tuple(m)
+
+"""
+    PModelConstants(FT)
+
+Creates a `PModelConstants` object with default values for the P-model constants.
+"""
+function PModelConstants(FT)
+    return PModelConstants(
+        R = LP.gas_constant(LP.LandParameters(FT)),
+        Kc25 = FT(39.97),
+        Ko25 = FT(27480),
+        To = FT(298.15),
+        ΔHkc = FT(79430),
+        ΔHko = FT(36380),
+        Drel = FT(1.6),
+        ΔHΓstar = FT(37830),
+        Γstar25 = FT(4.332),
+        Ha_Vcmax = FT(71513),
+        Hd_Vcmax = FT(200000),
+        aS_Vcmax = FT(668.39),
+        bS_Vcmax = FT(1.07),
+        Ha_Jmax = FT(49884),
+        Hd_Jmax = FT(200000),
+        aS_Jmax = FT(659.70),
+        bS_Jmax = FT(0.75),
+        Mc = FT(0.0120107),
+        oi = FT(0.2095),
+        aRd = FT(0.1012),
+        bRd = FT(-0.0005),
+        fC3 = FT(0.015),
+    )
+end
```

Comment:
If we write this function as follows, users will be able to change one parameter at a time while using the defaults for all the others. If there's only a subset you want to expose to users that you think they'll want to change, you can put those in the kwargs like I have here, and keep the others in the returned object like you have.

```suggestion
function PModelConstants{FT}(;
        R = LP.gas_constant(LP.LandParameters(FT)),
        Kc25 = FT(39.97),
        Ko25 = FT(27480),
        To = FT(298.15),
        ΔHkc = FT(79430),
        ΔHko = FT(36380),
        Drel = FT(1.6),
        ΔHΓstar = FT(37830),
        Γstar25 = FT(4.332),
        Ha_Vcmax = FT(71513),
        Hd_Vcmax = FT(200000),
        aS_Vcmax = FT(668.39),
        bS_Vcmax = FT(1.07),
        Ha_Jmax = FT(49884),
        Hd_Jmax = FT(200000),
        aS_Jmax = FT(659.70),
        bS_Jmax = FT(0.75),
        Mc = FT(0.0120107),
        oi = FT(0.2095),
        aRd = FT(0.1012),
        bRd = FT(-0.0005),
        fC3 = FT(0.015),
    )
    return PModelConstants{FT}(R, Kc25, ....)
end
```